tck/methods/account.go

```diff
@@ -0,0 +1,279 @@
+	Memo string `json:"memo"`
+}
+
+// UpdateAccount transaction to be associated with hedera.AccountUpdateTransaction and to be enough generic, so different
+// updates can be calling this func
+func (a *AccountService) UpdateAccount(_ context.Context, params UpdateAccountParams) (*AccountResponse, error) {
+	if params.NewPrivateKey != "" { // Proceed with UpdateAccountMemo
+		return a.UpdateAccountKey(params.AccountId, params.NewPublicKey, params.OldPrivateKey, params.NewPrivateKey)
+	} else if params.Memo != "" { // Proceed with UpdateAccountKey
+		return a.UpdateAccountMemo(params.AccountId, params.Key, params.Memo)
+	}
+	return nil, nil
+}
+
+// DeleteAccount deletes a provided account by signing the transaction with the key of that account
+func (a *AccountService) DeleteAccount(_ context.Context, params DeleteAccountParams) (*AccountResponse, error) {
+	accId, _ := hedera.AccountIDFromString(params.AccountId)
+	recipientId, _ := hedera.AccountIDFromString(params.RecipientId)
+	tx, err := hedera.NewAccountDeleteTransaction().SetGrpcDeadline(&threeSec).SetAccountID(accId).SetTransferAccountID(recipientId).FreezeWith(a.sdkService.Client)
+	if err != nil {
+		return nil, jrpc2.Errorf(-32603, "Internal error", ErrorStatus{Status: err.Error()})
+	}
+	signature, _ := hedera.PrivateKeyFromString(params.AccountKey)
+
+	resp, err := tx.Sign(signature).Execute(a.sdkService.Client)
+	if err != nil {
+		return nil, jrpc2.Errorf(-32603, "Internal error", ErrorStatus{Status: err.Error()})
+	}
+
+	receipt, _ := resp.GetReceipt(a.sdkService.Client)
+	return &AccountResponse{Status: receipt.Status.String()}, nil
+}
+
+// Helper methods
+// UpdateAccountKey updates an existing acoount id keys with provided params
+func (a *AccountService) UpdateAccountKey(accountId string, newPublicKey string, oldPrivateKey string, newPrivateKey string) (*AccountResponse, error) {
+	accId, _ := hedera.AccountIDFromString(accountId)
+	key, _ := hedera.PublicKeyFromString(newPublicKey)
+
+	tx, err := hedera.NewAccountUpdateTransaction().SetGrpcDeadline(&threeSec).SetAccountID(accId).SetKey(key).FreezeWith(a.sdkService.Client)
+
+	if err != nil {
+		return nil, jrpc2.Errorf(-32603, "Internal error", ErrorStatus{Status: err.Error()})
+	}
+	oldKey, _ := hedera.PrivateKeyFromString(oldPrivateKey)
+	newKey, _ := hedera.PrivateKeyFromString(newPrivateKey)
+	tx.Sign(oldKey)
+	tx.Sign(newKey)
+
+	resp, err := tx.Execute(a.sdkService.Client)
+	if err != nil {
+		return nil, jrpc2.Errorf(-32603, "Internal error", ErrorStatus{Status: err.Error()})
+	}
+
+	receipt, err := resp.GetReceipt(a.sdkService.Client)
+	if err != nil {
+		return nil, jrpc2.Errorf(-32603, "Internal error", ErrorStatus{Status: err.Error()})
+	}
+
+	return &AccountResponse{Status: receipt.Status.String()}, nil
+}
+
+// UpdateAccountMemo updates account memo of an existing account ID
+func (a *AccountService) UpdateAccountMemo(accountId string, key string, memo string) (*AccountResponse, error) {
```

Comment:
I see no reason why these should take multiple args. Would just pass the parameters of the caller function.

Reply:
Done